src/ape/managers/query.py

```diff
@@ -114,3 +114,55 @@ def query(self, query: QueryType, engine_to_use: Optional[str] = None) -> pd.Dat
             engine.update_cache(query, result)
 
         return result
+
+    def get_blocks(
```

Comment:
Actually... I would prefer this was different and that `stop_block` was the actual last block you want to get. And then the `range()` implementation on the `BlockContainer` was deal with that accordingly...

Reply:
I believe stop_block is the last block you want to get, it defaults to the last block of the chain, but you can call for a block you want

Reply:
If you look at the docs you copied in, it says `" The block number to stop before. Also the total
              number of blocks to get. If not setting a start value, is set by
              the first argument."`

This is because we wanted splicing behavior for the range method.

For this method, we want `stop_block` to be the last block you get. It does not currently work that way I don't think.

However, we still want it to work that way from the `BlockContainer.range()` perspective... So some refactoring is still needed

Reply:
ohhh yes, I follow, I see what you mean. I think if I change that though, it will break all of these tests

Reply:
Make `chain_manager.blocks.range()` work exactly like it did before EXCEPT it now utilizes the `QueryManager` (which does the heavy lifting via most of the logic you copy in). The one difference is that in query_manager, make `stop_block` refer to the actual last block you want to get.

So it won't break the `chain_manager.range()` tests because we still want that to mimic the python splice behavior.

And then add new tests for this method in the `QueryManager`.